File: C++/035. Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int size = nums.size(), left, right, center;
        left = 0;
        right = size-1;
        while (left<=right)
        {
            if (left==right)
            {
                if (nums[left]==target){return left;}
                else{return (nums[left]<target)? left+1:left;}
            }
            else
            {
                center = (left+right)/2;
                if(nums[center]<target)
                {
                    left=center+1;
                }
                else if(nums[center]>target)
                {
                    right = center-1;
                }
                else
                {
                    return center;
                }
            }
        }
        return center;
    }
};

File: python/035. Search Insert Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n = len(nums)
        if n == 0: return 0
        left, right = 0, n-1
        while left <= right:
            mid = (left + right) / 2
            if nums[mid] >= target:
                right = mid - 1
            else:
                left = mid + 1
        return left